test/ibmq/test_ibmq_job_attributes.py

```diff
@@ -107,6 +107,37 @@ def test_job_name(self, provider):
         self.assertEqual(len(retrieved_jobs), 1)
         self.assertEqual(job_id, retrieved_jobs[0].job_id())
 
+    @requires_provider
+    def test_job_name_update(self, provider):
+        """Test changing the name associated with a job."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+
+        # Use a unique job name
+        initial_job_name = str(time.time()).replace('.', '')
+        job = backend.run(qobj, job_name=initial_job_name, validate_qobj=True)
+        job_id = job.job_id()
+        # TODO No need to wait for job to run once api is fixed
+        while job.status() not in JOB_FINAL_STATES + (JobStatus.RUNNING,):
+            time.sleep(0.5)
+        rjob = provider.backends.retrieve_job(job_id)
+        self.assertEqual(rjob.name(), initial_job_name)
```

Comment:
Setting a job name with `backend.run()` is already covered in another test case; no need to verify this here unless you want to combine the 2 test cases.